micro-rdk/src/esp32/entry.rs

```diff
@@ -337,55 +369,105 @@ async fn serve_async_with_external_network<S>(
     exec: Esp32Executor,
     info: Option<ProvisioningInfo>,
     storage: S,
-    repr: RobotRepresentation,
+    mut repr: RobotRepresentation,
     network: impl Network,
     max_webrtc_connection: usize,
 ) -> Result<(), Box<dyn std::error::Error>>
 where
-    S: RobotCredentialStorage + WifiCredentialStorage + Clone + 'static,
-    <S as RobotCredentialStorage>::Error: Debug,
-    ServerError: From<<S as RobotCredentialStorage>::Error>,
+    S: RobotConfigurationStorage + WifiCredentialStorage + Clone + 'static,
+    <S as RobotConfigurationStorage>::Error: Debug,
+    ServerError: From<<S as RobotConfigurationStorage>::Error>,
     <S as WifiCredentialStorage>::Error: Sync + Send + 'static,
 {
     use crate::common::provisioning::server::serve_provisioning_async;
 
     use super::conn::mdns::Esp32Mdns;
 
+    let mut client_connector = Esp32TLS::new_client();
     let info = info.unwrap_or_default();
     let mut last_error: Option<Box<dyn std::error::Error>> = None;
 
-    loop {
-        // Credentials are present let's check we can connect
-        if storage.has_stored_credentials() {
-            let validated = loop {
-                // should check internet when implementing Cached Config
-                if let Err(error) = network.is_connected() {
-                    log::info!(
-                        "Externally managed network, not connected yet cause {:?}",
-                        error
-                    );
-                    Timer::after(Duration::from_secs(3)).await;
-                    continue;
+    let app_client = 'provisioned: loop {
+        if storage.has_robot_credentials() {
+            log::info!("Found cached robot credentials; attempting to serve");
+
+            if storage.has_robot_configuration() {
+                if let RobotRepresentation::WithRegistry(ref registry) = repr {
+                    log::info!("Found cached robot configuration; speculatively building robot from config");
+                    match LocalRobot::from_cloud_config(
```

Comment:
feel's like this might be better placed outside the loop? Also i am not sure i am reading this correctly but there is a case where when the robot config is cached, the esp32 starts the robot would be created based on the cached config so if a newer config exists it won't be applied? And the time would not be properly set?

Reply:
Sorry: hit the button too soon. If you want, I can try to swap the loop around to do a connectivity check first and then only build from cache on failure. Now that I look at the loop it may not be that hard. Let me know if you feel strongly enough about it that you want me to fold it in for this review. Alternatively, we could lock in the current gains and work on that in a subsequent iteration.

Reply:
I just wanted to confirm I was reading this correctly, i don't think we should the current behavior. And it makes sense to restart & rebuild once we have config checks

Reply:
Just checking, but do you mean "don't think we should **change** the current behavior"?